src/index.js

```diff
@@ -230,6 +233,15 @@ var InputPassword = React.createClass({
     // allow onChange to be passed from parent and not override default prop
     var {onChange, ...props} = this.props;
 
+    // overcome problem with firefox resetting the input selection point
+    var that = this;
+    setTimeout(function() {
+      if (!/Firefox/.test(navigator.userAgent)) return;
+      var elem = that.refs[that.props.id].getDOMNode();
```

Comment:
id not required. Maybe this should be:

var elem = that.refs[that.props.id || "input"].getDOMNode();